### ISSUE TITLE
Add ati-smarttag and google-ima3 redirect resources

### ISSUE THREAD

**Issue by Yuki2718** [uBlockOrigin/uBlock-issues#2158]
### Prerequisites

- [X] I verified that this is not a filter issue (MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues))
- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### Description

AG implementation:
https://github.com/AdguardTeam/Scriptlets/blob/master/wiki/about-redirects.md#ati-smarttag
https://github.com/AdguardTeam/Scriptlets/blob/master/wiki/about-redirects.md#-%EF%B8%8F-google-ima3

They often cause problems and exception have been made each time. If the redirect resources work without problem, mayb we can generally redirect and reduce the freuqency of those problems. Even if not, at least we can reduce exposure to these serveces by domain-specific redirect rules with `$important`.

### A specific URL where the issue occurs

https://www.rte.ie/news/business/2022/0626/1306958-one-third-of-households-just-making-ends-meet/

### Steps to Reproduce

1. For ati-smarttag visit the URL, see embedded player is working with default setting thx to exception.
2. If you add `||tag.aticdn.net/$script,domain=rte.ie,important`, the player will be broken.
3. If you visit the URL with AG, you'll see working player without exception (as I added the rule in https://github.com/AdguardTeam/AdguardFilters/commit/cb87af87422c8414300a3cf7a1860f302493501d)

### Expected behavior

NA

### Actual behavior

Exception is often made for these two.

### uBlock Origin version

1.43.1b3

### Browser name and version

Firefox 101.0.1

### Operating System and version

Windows 10

**Comment by krystian3w** [comment]
Duplicate of topic from hidden volunteer?

**Comment by krystian3w** [comment]
I see typo: <q>, mayb we</q>

**Comment by krystian3w** [comment]
https://github.com/AdguardTeam/Scriptlets/issues/244

https://github.com/AdguardTeam/Scriptlets/issues/239

**Comment by gorhill** [comment]
Firefox has a shim for ima3: <https://searchfox.org/mozilla-central/source/browser/extensions/webcompat/shims/google-ima.js#6>.

I should just borrow it I guess. It would be nice to have these shims exposed to any API so that extensions can use them.

**Comment by Yuki2718** [comment]
Looks compatible: https://www.gnu.org/licenses/license-list.en.html#MPL-2.0

**Comment by krystian3w** [comment]
@mozilla fyi!

**Comment by gorhill** [comment]
I can't use Mozilla's version as is it turns out because it uses JS syntax ([private properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields) that is valid starting with Chromium 74/Firefox 90 -- so I will have to modify to avoid relying on this private field syntax.

Additionally, I used this test case where I recently excepted ima3 to bypass anti-blocker to validate that it works:

- Navigate to `https://www.novelgames.com/en/saliclawsolitaire/`
- Create a block rule for `imasdk.googleapis.com` to override the exception filter
- Reloda the tab
- Click play button

Without shim, anti-blocker message.

Add filter (need priority to win against the existing redirect in uBO's lists):

    ||imasdk.googleapis.com/js/sdkloader/ima3.js$3p,redirect-rule=google-ima.js:1

Reload and click play button: still an anti-blocker message. There is a missing event in Firefox's version, and when I add this missing event, the shim works.

**Comment by gorhill** [comment]
@wisniewskit

For your interest, I had to [add the event `CONTENT_RESUME_REQUESTED` in `start()`](https://github.com/gorhill/uBlock/commit/067e128163be175377fa094054b881c0f3f5fe92?diff=unified#diff-f14b39592541bfc46cd8cd4cb2b6a106e81fd886f407c6d338c033e8248832deR428) to unbreak the case [detailed above](https://github.com/uBlockOrigin/uBlock-issues/issues/2158#issuecomment-1242975050).

**Comment by wisniewskit** [comment]
@gorhill, thanks for letting me know, and sorry for the extra porting work!

**Comment by Yuki2718** [comment]
Steps to reproduce (or steps to just watch video) is very tedious, but in `https://github.com/easylist/easylist/issues/16535` I found redirecting causes pausing video not to work. Was there any similar problem?

**Comment by stephenhawk8054** [comment]
> Steps to reproduce (or steps to just watch video) is very tedious, but in [easylist/easylist#16535](https://github.com/easylist/easylist/issues/16535) I found redirecting causes pausing video not to work. Was there any similar problem?

Yes, redirecting google-ima sometimes still cause breakages. Looks like each site can have slightly different coding with it. Some other discussions here: https://github.com/uBlockOrigin/uBlock-issues/issues/2265

**Comment by Yuki2718** [comment]
> Does your console throw any errors related to it?

I don't see anything related to imasdk.
